ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala

```diff
@@ -419,4 +419,12 @@ class InMemoryLedger(
         }
       }
     }
+
+  override def removeExpiredDeduplicationData(currentTime: Instant): Future[Unit] =
+    Future.successful {
+      this.synchronized {
+        commands.filter(p => p._2.deduplicateUntil.isAfter(currentTime))
```

Comment:
This doesn't actually change the underlying commands though. Maybe you wanted to use `retain`?